core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
It seems better to match what x86 does: move handling of raw-bits-valid instrs up above this decode_info_t init.  That will be more efficient and more symmetric.  Then we have an invariant that it's level 3+ by the time we get here, which the operands_valid assert will double-check.